tests/functional/event_handler/test_api_gateway.py

```diff
@@ -715,6 +715,42 @@ def get_network_account(account_id: str, network_id: str):
     app.resolve(event, {})
 
 
+def test_similar_dynamic_routes_with_whitespaces():
+    # GIVEN
+    app = ApiGatewayResolver()
+    event = deepcopy(LOAD_GW_EVENT)
+
+    # WHEN
+    # r'^/accounts/(?P<account_id>\\w+\\b)$' # noqa: E800
+    @app.get("/accounts/<account_id>")
+    def get_account(account_id: str):
+        assert account_id == "single account"
+
+    # r'^/accounts/(?P<account_id>\\w+\\b)/source_networks$' # noqa: E800
+    @app.get("/accounts/<account_id>/source_networks")
+    def get_account_networks(account_id: str):
+        assert account_id == "nested account"
+
+    # r'^/accounts/(?P<account_id>\\w+\\b)/source_networks/(?P<network_id>\\w+\\b)$' # noqa: E800
+    @app.get("/accounts/<account_id>/source_networks/<network_id>")
+    def get_network_account(account_id: str, network_id: str):
+        assert account_id == "nested account"
+        assert network_id == "network 123"
+
+    # THEN
+    event["resource"] = "/accounts/{account_id}"
+    event["path"] = "/accounts/single account"
+    app.resolve(event, None)
+
+    event["resource"] = "/accounts/{account_id}/source_networks"
+    event["path"] = "/accounts/nested account/source_networks"
+    app.resolve(event, None)
+
+    event["resource"] = "/accounts/{account_id}/source_networks/{network_id}"
+    event["path"] = "/accounts/nested account/source_networks/network 123"
+    app.resolve(event, {})
```

Comment:
Because the whitespace is a special one, it'd be great for us to assert on status code too - I noticed we missed in the other similar dynamic routes test.

To speed up the release, I'm gonna commit the suggestion and wait for CI to confirm before merging it

```suggestion
    # THEN
    event["resource"] = "/accounts/{account_id}"
    event["path"] = "/accounts/single account"
    assert app.resolve(event, {})["statusCode"] == 200

    event["resource"] = "/accounts/{account_id}/source_networks"
    event["path"] = "/accounts/nested account/source_networks"
    assert app.resolve(event, {})["statusCode"] == 200
    
    event["resource"] = "/accounts/{account_id}/source_networks/{network_id}"
    event["path"] = "/accounts/nested account/source_networks/network 123"
    assert app.resolve(event, {})["statusCode"] == 200
```